### DISCUSSION TITLE
Supporting the project by implementing Spotify Design Guideline

### DISCUSSION THREAD

**Discussion by zzikkzzakk** [gilbarbara/react-spotify-web-playback#103] [General]
In order to comply to Spotify Design Guidelines a Spotify icon would need to be added to the player.
See "Playing Views" in the [guidelines](https://developer.spotify.com/documentation/general/design-and-branding/).

I would try to implement it and create a pull request, but I'm lacking information how to setup the development environment for this project.

Any hints would be appreciated

**Comment by gilbarbara** [comment]
hey @zzikkzzakk 

I'm planning to update the UI for v1. Take a look at this [issue](https://github.com/gilbarbara/react-spotify-web-playback/issues/41).
With that in mind, I don't think changing the current layout would be a good idea. Feel free to suggest some ideas.

**Reply by zzikkzzakk** [reply]
Hey @gilbarbara , great to hear you are already working on a UI update. 
In the mean time for submitting my web app to Spotify I would need the icon. Could you explain how to set up the development environment and get the demo running with local code?
We could discuss this in a seperate thread if it's not straight forward.

**Reply by gilbarbara** [reply]
Install the demo dependencies and load the library from `../src` instead.

Just to be clear, I won't accept a PR with UI changes for the current layout

**Reply by zzikkzzakk** [reply]
Thanks for the hint, I will try it. Understand your last point 👍

**Reply by zzikkzzakk** [reply]
These were the steps for my development setup:

- In package.json of demo folder:
    - Change dependency of react-spotify-web-playback to local filesystem, i.e. `"react-spotify-web-playback": "file:.."` 
    - Upgrade dependency `"react-scripts": "^5.0.1"`
    - add eslint-config to devDependencies: `"@gilbarbara/eslint-config": "^0.2.4"` 
- In root folder of react-spotify-web-playback: `npm install` 
- In root folder: `npm run build`
- In demo folder: `npm install`

**Comment by gilbarbara** [comment]
@zzikkzzakk I've implemented Spotify attribution in version 0.12.0, if you still need it
